### PR TITLE
python 3 only by pyupgrade

### PR DESCRIPTION
**Pull request by pgajdos**
Given python 3.6+ support is declared, I guess e. g. six do not need to be required.

### CODE REVIEW

**Comment by a-detiste** [comment]
Yes, `six` should indeed be removed ...everywhere ...: https://wiki.debian.org/Python3-six-removal

File: memcache.py [hunk 1]
```diff
@@ -507,7 +504,7 @@ def delete_multi(self, keys, time=None, key_prefix='', noreply=False):
                 write(cmd)
             try:
                 server.send_cmds(b''.join(bigcmd))
-            except socket.error as msg:
+            except OSError as msg:
```
Comment (linsomniac) [inline memcache.py:507]: Can you point me at a recommendation to replace socket.error with OSError?  Looks like in the 3.12 source code they document socket.error as an alias for OSError, so this is correct, but I'm reluctant to change it without seeing where the recommendation comes from.

File: memcache.py [hunk 2]
```diff
@@ -522,11 +519,11 @@ def delete_multi(self, keys, time=None, key_prefix='', noreply=False):
         for server in dead_servers:
             del server_keys[server]
 
-        for server, keys in six.iteritems(server_keys):
+        for server, keys in server_keys.items():
             try:
                 for key in keys:
                     server.expect(b"DELETED")
-            except socket.error as msg:
+            except OSError as msg:
```
Comment (linsomniac) [inline memcache.py:526]: (Ditto)

File: memcache.py [hunk 3]
```diff
@@ -557,8 +554,8 @@ def delete(self, key, noreply=False):
             line = server.readline()
             if line and line.strip() in [b'DELETED', b'NOT_FOUND']:
                 return 1
-            self.debuglog('delete expected DELETED or NOT_FOUND, got: %r' % (line,))
-        except socket.error as msg:
+            self.debuglog('delete expected DELETED or NOT_FOUND, got: {!r}'.format(line))
+        except OSError as msg:
```
Comment (linsomniac) [inline memcache.py:558]: (Ditto)

File: memcache.py [hunk 4]
```diff
@@ -593,8 +590,8 @@ def touch(self, key, time=0, noreply=False):
             line = server.readline()
             if line and line.strip() in [b'TOUCHED']:
                 return 1
-            self.debuglog('touch expected TOUCHED, got: %r' % (line,))
-        except socket.error as msg:
+            self.debuglog('touch expected TOUCHED, got: {!r}'.format(line))
+        except OSError as msg:
```
Comment (linsomniac) [inline memcache.py:594]: (Ditto)

File: memcache.py [hunk 5]
```diff
@@ -668,7 +665,7 @@ def _incrdecr(self, cmd, key, delta, noreply=False):
             if line is None or line.strip() == b'NOT_FOUND':
                 return None
             return int(line)
-        except socket.error as msg:
+        except OSError as msg:
```
Comment (linsomniac) [inline memcache.py:668]: (Ditto)

File: memcache.py [hunk 6]
```diff
@@ -939,7 +936,7 @@ def set_multi(self, mapping, time=0, key_prefix='', min_compress_len=0,
                     else:
                         notstored.append(prefixed_to_orig_key[key])
                 server.send_cmds(b''.join(bigcmd))
-            except socket.error as msg:
+            except OSError as msg:
```
Comment (linsomniac) [inline memcache.py:939]: (Ditto)

File: memcache.py [hunk 7]
```diff
@@ -957,15 +954,15 @@ def set_multi(self, mapping, time=0, key_prefix='', min_compress_len=0,
...
+        for server, keys in server_keys.items():
             try:
                 for key in keys:
                     if server.readline() == b'STORED':
                         continue
                     else:
                         # un-mangle.
                         notstored.append(prefixed_to_orig_key[key])
-            except (_Error, socket.error) as msg:
+            except (_Error, OSError) as msg:
```
Comment (linsomniac) [inline memcache.py:965]: (Ditto)

File: memcache.py [hunk 8]
```diff
@@ -1054,7 +1051,7 @@ def _unsafe_set():
                 if noreply:
                     return True
                 return server.expect(b"STORED", raise_exception=True) == b"STORED"
-            except socket.error as msg:
+            except OSError as msg:
```
Comment (linsomniac) [inline memcache.py:1054]: (Ditto)(

File: memcache.py [hunk 9]
```diff
@@ -1067,7 +1064,7 @@ def _unsafe_set():
             try:
                 if server._get_socket():
                     return _unsafe_set()
-            except (_ConnectionDeadError, socket.error) as msg:
+            except (_ConnectionDeadError, OSError) as msg:
```
Comment (linsomniac) [inline memcache.py:1067]: (Ditto)

File: memcache.py [hunk 10]
```diff
@@ -1105,7 +1102,7 @@ def _unsafe_get():
                     value = self._recv_value(server, flags, rlen)
                 finally:
                     server.expect(b"END", raise_exception=True)
-            except (_Error, socket.error) as msg:
+            except (_Error, OSError) as msg:
```
Comment (linsomniac) [inline memcache.py:1105]: (Ditto)

File: memcache.py [hunk 11]
```diff
@@ -1121,7 +1118,7 @@ def _unsafe_get():
                 if server.connect():
                     return _unsafe_get()
                 return None
-            except (_ConnectionDeadError, socket.error) as msg:
+            except (_ConnectionDeadError, OSError) as msg:
```
Comment (linsomniac) [inline memcache.py:1121]: (Ditto)

File: memcache.py [hunk 12]
```diff
@@ -1201,11 +1198,11 @@ def get_multi(self, keys, key_prefix=''):
 
         # send out all requests on each server before reading anything
         dead_servers = []
-        for server in six.iterkeys(server_keys):
+        for server in server_keys.keys():
             try:
                 fullcmd = b"get " + b" ".join(server_keys[server])
                 server.send_cmd(fullcmd)
-            except socket.error as msg:
+            except OSError as msg:
```
Comment (linsomniac) [inline memcache.py:1205]: (Ditto)

File: memcache.py [hunk 13]
```diff
@@ -1227,7 +1224,7 @@ def get_multi(self, keys, key_prefix=''):
                         # un-prefix returned key.
                         retvals[prefixed_to_orig_key[rkey]] = val
                     line = server.readline()
-            except (_Error, socket.error) as msg:
+            except (_Error, OSError) as msg:
```
Comment (linsomniac) [inline memcache.py:1227]: (Ditto)

File: memcache.py [hunk 14]
```diff
@@ -1410,7 +1407,7 @@ def _get_socket(self):
         except socket.timeout as msg:
             self.mark_dead("connect: %s" % msg)
             return None
-        except socket.error as msg:
+        except OSError as msg:
```
Comment (linsomniac) [inline memcache.py:1410]: (Ditto)

File: memcache.py (hunk 1 above)
Comment (linsomniac) [reply memcache.py:507]: Scratch that, I found where it is deprecated.  Good spot!

**Comment by linsomniac** [comment]
> Given python 3.6+ support is declared, I guess e. g. six do not need to be included

This looks really good, and I found the deprecation of socket.error, so I'm going to merge this, thank you!

**Pull request merged**